test/framework/labels.go

```diff
@@ -18,11 +19,31 @@ func ValidateControlPlaneLabels(controlPlane v1alpha1.ControlPlaneConfiguration,
 	return validateLabels(controlPlane.Labels, node)
 }
 
+// ValidateControlPlaneFailureDomainLabels validate if Cloudstack provider replaces ds.meta_data.failuredomain with proper failuredomain name
+// in control plane node label 'cluster.x-k8s.io/failure-domain'.
+func ValidateControlPlaneFailureDomainLabels(controlPlane v1alpha1.ControlPlaneConfiguration, node corev1.Node) error {
+	if controlPlane.MachineGroupRef.Kind == "CloudStackMachineConfig" {
+		logger.V(4).Info("Validating control plane node failuredomain label")
+		return validateFailureDomainLabel(controlPlane.Labels, node)
+	}
+	return fmt.Errorf("ds.meta_data.failuredomain placeholder in node label is currently only supported in CloudStack provider")
+}
+
 func ValidateWorkerNodeLabels(w v1alpha1.WorkerNodeGroupConfiguration, node corev1.Node) error {
 	logger.V(4).Info("Validating worker node labels", "worker node group", w.Name)
 	return validateLabels(w.Labels, node)
 }
 
+// ValidateWorkerNodeFailureDomainLabels validate if Cloudstack provider replaces ds.meta_data.failuredomain with proper failuredomain name
+// in worker group node label 'cluster.x-k8s.io/failure-domain'.
+func ValidateWorkerNodeFailureDomainLabels(w v1alpha1.WorkerNodeGroupConfiguration, node corev1.Node) error {
+	if w.MachineGroupRef.Kind == "CloudStackMachineConfig" {
```

Comment:
I think we have a constant for this, could we use it here?

Reply:
updated to use constant.